### PR TITLE
Properly visit FunctionExpression; handle ArgumentInvocation

### PR DESCRIPTION
**Pull request by srawlins**
Also refactor rule tests.

FunctionExpression is needed for upcoming rule for #24. Implementing it and visiting the ArgumentInvocation makes other rules more complete.

The tests were using a single `_DummyRule` that caught numbers and booleans, to test whether Rule indeed visited down the tree and reported back up. However, to test all of the components of an ArgumentInvocation, I wanted to also report on variables. This spurred a whole refactoring of these tests, better isolating what is being tested (before, I had to be careful about placing numbers and booleans in the same Sass script; this refactoring improves the expectation on line 233).

### CODE REVIEW

File: lib/src/rule.dart [hunk 1]
```diff
@@ -266,6 +267,20 @@ abstract class Rule
     throw new UnimplementedError();
   }
 
+  /// Visit [node], an ArgumentInvocation, returning the lint found within.
+  List<Lint> _visitArgumentInvocation(ArgumentInvocation node) {
+    var lint = <Lint>[];
+    for (var argument in node.positional) {
+      lint.addAll(argument.accept(this));
+    }
+    node.named.forEach((String name, Expression value) {
```
Comment (nex3) [inline lib/src/rule.dart]: Nit: don't type-annotate arguments that can be inferred.

File: test/rule_test.dart [hunk 2]
```diff
@@ -8,30 +8,65 @@ import 'package:sass_linter/src/linter.dart';
 import 'package:sass_linter/src/rule.dart';
 import 'package:test/test.dart';
 
-/// A rule which tries to emulate the shape of a typical lint rule.
+/// A linter rule which reports lint on every boolean literal.
 ///
-/// Testing with this rule can verify that lint found by a rule which only
-/// overrides a tiny set of `visit` methods is reported all the way up the tree.
-class _DummyRule extends Rule {
```
Comment (nex3) [inline test/rule_test.dart]: What if you kept `_DummyRule`, but gave it named parameters that controlled exactly what lint it reported? E.g. `_DummyRule(booleans: true)`, `_DummyRule(numbers: true)`. That might reduce boilerplate.

File: test/rule_test.dart [hunk 3]
```diff
@@ -219,8 +252,73 @@ void main() {
...
+
+    var lint = lints.single;
+    expect(lint.line, 1);
+    expect(lint.column, 24);
+  });
+
+  test('reports lint found within a function invocation keyword varargs', () {
+    var lints = getVariableLints(r'''
+        $args: (color: red, amount: 10%);
+        $my-red: darken($args...);
```
Comment (nex3) [inline test/rule_test.dart]: `$args` here is just the normal varargs. The difference between normal varargs and keyword varargs is purely syntactic. You only get keyword varargs when you pass two varargs, as in `darken($args..., $kwargs...)`.

Comment (srawlins) [reply test/rule_test.dart]: Done.

File: test/rule_test.dart (hunk 2 above)
Comment (srawlins) [reply test/rule_test.dart]: Good idea. Done.

File: lib/src/rule.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/rule.dart]: Done.

File: test/rule_test.dart [hunk 4]
```diff
@@ -8,30 +8,58 @@ import 'package:sass_linter/src/linter.dart';
 import 'package:sass_linter/src/rule.dart';
 import 'package:test/test.dart';
 
-/// A rule which tries to emulate the shape of a typical lint rule.
+/// A configurable linter rule.
 ///
-/// Testing with this rule can verify that lint found by a rule which only
-/// overrides a tiny set of `visit` methods is reported all the way up the tree.
+/// This rule only overrides a few leaf visit method. This rule can be used in
```
Comment (nex3) [inline test/rule_test.dart]:
```suggestion
/// This rule only overrides a few leaf visit methods. This rule can be used in
```

**Pull request merged**